src/core/transform.js

```diff
@@ -129,21 +132,28 @@ export default class Transform {
       feedbackMap = feedbackMap || sourceDestinationMap;
     }
 
-    assert(sourceBuffers && vs && Array.isArray(varyings) && elementCount >= 0);
+    assert(sourceBuffers && vs && elementCount >= 0);
     // If feedbackBuffers are not provided, sourceDestinationMap must be provided
     // to create destinaitonBuffers with layout of corresponding source buffer.
     assert(feedbackBuffers || feedbackMap, ' Transform needs feedbackBuffers or feedbackMap');
     for (const bufferName in feedbackBuffers || {}) {
       assert(feedbackBuffers[bufferName] instanceof Buffer);
     }
 
+    // If varyings are not provided feedbackMap must be provided to build varyings
+    assert(Array.isArray(varyings) || feedbackMap);
+    let varyingsArray = varyings;
+    if (!Array.isArray(varyings)) {
+      varyingsArray = Object.values(feedbackMap);
```

Comment:
Nit: You could just do `varyings = Object.values...` and not declare a new var.

Reply:
Just avoiding changing object passed to this method.